pkg/cvo/cvo.go

```diff
@@ -410,24 +407,35 @@ func (optr *Operator) rememberLastUpdate(config *configv1.ClusterVersion) {
 func (optr *Operator) getOrCreateClusterVersion() (*configv1.ClusterVersion, bool, error) {
 	obj, err := optr.cvLister.Get(optr.name)
 	if err == nil {
-		return obj, false, nil
+		// if we are waiting to see a newer cached version, just exit
+		if optr.isOlderThanLastUpdate(obj) {
+			return nil, true, nil
+		}
+
+		// ensure that the object we do have is valid
+		errs := validation.ValidateClusterVersion(obj)
+		changed, err := optr.syncInitialObjectStatus(obj, errs)
+		if err != nil {
+			return nil, false, err
+		}
+
+		// for fields that have meaning that are incomplete, clear them
+		// prevents us from loading clearly malformed payloads
+		obj = validation.ClearInvalidFields(obj, errs)
```

Comment:
Is this mutating lister cache?

Reply:
we deep copy before making a change in that method.